data-prepper-plugins/mongodb/src/test/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorkerTest.java

```diff
@@ -380,4 +382,154 @@ void test_processStream_terminateChangeStreamSuccess() {
         verify(failureItemsCounter, never()).increment();
         verify(mockPartitionCheckpoint).resetCheckpoint();
     }
+
+    @Test
+    void test_processStream_dataTypeConversionSuccess() {
```

Comment:
Suggest using separate test cases for different datatypes to make this more debuggable.

Reply:
That's a great point! It will also help maintain the code as we make changes.

@dinujoh , I see a couple of options:

1. Create a ParameterizedTest
2. Create an `@Nested` class which has the shared setup. Then create small tests for each data type.

Reply:
Will Create a ParameterizedTest